packages/material-ui-lab/src/SliderStyled/SliderStyled.js

```diff
@@ -436,6 +436,10 @@ Slider.propTypes = {
    * @default 1
    */
   step: PropTypes.number,
+  /**
+   * Common system props.
+   */
+  system: PropTypes.object,
```

Comment:
We should probably come up with better name here... But I would definitely vote for defining all system props under one prop instead of spreading them on each component, mainly because of the following reasons:
- we are already specifying all styles props under one prop - `stylesProps`
- we would avoid props explosion
- all system props would be easily discoverable

Reply:
`systemProps` for consistency?

Reply:
I was thinking more of a prop like `styles`, or `css` (css will conflict with emotion's css), but even `systemProps` sounds better :)

Reply:
Let's get some other feedback before changing it.

Reply:
Should we have a 📊 on Twitter about the best API for the system? styled-system push for flattening; theme-ui.com pushes for isolation under one prop. Both have the same author.

Reply:
Seems like people are more in favor of spreading all system props - 
![image](https://user-images.githubusercontent.com/4512430/94901311-c0222200-0496-11eb-804e-aab6bae590e6.png)

Should we go with that approach? We will have to figure out how to handle duplicate props (color is one I can think of)

Reply:
I'm not sure how valid the result is, as it simply asks "which API looks nicer". I doubt most respondents gave much thought to the broader implications of each approach WRT:

- prop collisions
- type definitions
- code completion
- documentation
- code size / complexity
- more?

Reply:
I have the same feeling, they just voted for what looked better, without really thinking about it... I was planning to respond on this on Monday and ask for more feedback on why they voted as they did, or maybe give our reasoning on props/cons for the APIs. I also plan to do a benchmark for having one prop vs many just to double check if there will be some perf differences as well.

Reply:
We were discussing with @oliviertassinari also about prefixing the system props with `$` or `css` so that they won't conflict with the component's props

Reply:
I had a look at the answers on Twitter, it seems that we have the following pros for each option:

Option 1 pros:

- More popular (answer: What should be the API be?) But we didn't ask with $ as a prefix.
- Won't need to introduce a BCs on the existing Box component.
- Doesn't invalidate React.memo.

Option 2 pros:

- Allows supporting a superset of CSS (not a subset like we have in v4), e.g #15878.
- (to be confirmed and how much?) Better performance by 1. allowing to skip the whole system logic if the prop is not used, 2. having a more efficient prop forwarding logic.
- Allows simpler code migration with styled().
- Improve TypeScript autocomplete.

For confirmation, the problems we are going after are?

1. Naming things is hard
2. Jumping between JS and CSS in your editor wastes time (still true when using `styled`)
3. Inline the styles, no specificity war
4. More velocity when writing styles, fewer boilerplate of code to move around
5. Increase usage of design tokens

Looking at the analytics, we have the following popularity per system page:

<img width="360" alt="Capture d’écran 2020-10-03 à 14 36 27" src="https://user-images.githubusercontent.com/3165635/94991829-e07cda00-0585-11eb-8d11-4a05c183c360.png">

The primary use case seems for the system (so far) seems to be about layout.

How about?

- The Box and core components support a `sx` prop, which is a superset of CSS:

```jsx
    <Slider
      sx={{
        p: 1,
        backgroundColor: 'primary.main',
        ':hover': {
          backgroundColor: 'red',
        },
      }}
    >
```

I have seen libraries support the `css` prop too, I don't see a use case. Should we ignore it? https://codesandbox.io/s/react-codesandbox-forked-vyg9h?file=/src/index.js. Using the name `sx` helps with interoperability and past learning of the community.

- The Box component accepts flatten CSS properties, it's a style helper. It's a subset of CSS. We don't add the $ prefix.

```jsx
<Box p={6}>
```

- We remove the `clone` prop. It's no longer needed since core components will support the system.
- We remove/rename the `css` prop to `sx` https://material-ui.com/system/basics/#css-property.

Reply:
What is `sx` supposed to mean?

Though I do favor the single prop over spreading. Makes it easier to avoid collisions and aids static analysis.

Reply:
> What is sx supposed to mean?

@eps1lon I don't know. I interpret "s" as "system" or "styles". I interpret "x" as xml (like JSX) or a a placeholder for any value with superpowers. It has the advantage to be already known by Theme-UI's users. <kbd>s</kbd> and <kbd>x</kbd> are also close on the AZERTY and QWERTY keyboards (same location).

Reply:
That doesn't sound like it's intuitive at all. What's wrong with `system`?

Reply:
I don't have a strong preference either way. I think that both are fine. The advantages I can think of with `sx` over `system`:

- It announces what the system is about upfront: save time with arbitrary conventions and shorthands, e.g. two letters. If we go back to pain points the system is meant to solve (shared in https://github.com/mui-org/material-ui/pull/22819#discussion_r499146934), I believe it's a lot about being able to write CSS faster. While we usually say that developers spend x10 more time reading code than writing it (in general). I think that for styles, the balance is more round 50/50. At least it's my experience so far, instead of reading styles, I let the browser do it for me as often as possible.
- It's already known by Theme-UI's users, no learning step, e.g. it's similar to using `mt` already learned by Bootstrap/Tailwind/styled-system's users.

Reply:
I think you structured this backwards. Are you saying that `sx` announces upfront what it is about but `system` doesn't? I don't think this will hold true at all. Especially considering we have a lot of users with a background in backend development.

As to writing speed: You write this prop once.

Reply:
Or do you mean to say that `sx` is about announcing that the system about unreadable prop names? :wink:

Reply:
> Or do you mean to say that sx is about announcing that the system is about unreadable prop names? 😉

Yes, that 😁

I propose we wait for Marija point of view. I think that we can settle the matter with whatever she thinks is best.

Reply:
Maybe we can consider `css` as a candidate for the name of the prop. I have seen it used in https://react-ui.dev/core-concepts/ideas#Element%20primitive.

Reply:
From my point of view, `system` is the easiest to remember prop name that makes sense. We are documenting this as system on our docs, I don't see why we should invent different name for it to be honest...

On the `sx` I like the name and I think it is good and well known, but if I see it I would think that it should work on all element using custom pragma for the  jsx transofrmer `/** @jsx jsx */` - as this is usually how it is implemented in other libraries

`css` is good name, but I would avoid as it is already used by emotion which is our recommended default style engine. Also it does not say that it is anyway related to the system.

Reply:
> and well known

We need to be careful with that statement and consider audiences. A very large portion are have a backend background. So far I've only heard of it in the context of Theme-UI.

Reply:
So `system` > `sx` > `css`?

Reply:
Regarding the flatten vs prop tradeoff. Any thoughts [on it](https://github.com/mui-org/material-ui/pull/22819#discussion_r499146934)? The summarized version:

- Box component: subset of CSS, flattened system props + superset of CSS, `system` prop
- Most core components: superset of CSS, `system` prop only
- Some core components that are responsible for accessing design tokens or doing layout have flattened system prop: Grid, Stack, Typography (e.g. fontWeight).

Reply:
> So system > sx > css?

Yep, and I am saying `system` in case if we are going to use it as a simply collection of system props, if we are going to treat is as a superset of CSS, then I would rather use a prop like `styles` or `sx` for it... In my opinion there is a big difference between these two use-cases

> Regarding the flatten vs prop tradeoff. Any thoughts on it? The summarized version:
> 
> Box component: subset of CSS, flattened system props + superset of CSS, system prop

Agree, again if we are using the system props as superset of CSS I am revoking the system `prop` as a name, as it is not system anymore is more styles/css. 😄 

> Most core components: superset of CSS, system prop only

Agree 👍 

> Some core components that are responsible for accessing design tokens or doing layout have flattened system prop: Grid, Stack, Typography (e.g. fontWeight).

These should basically be the same as `Box` then, both flattened system prop + `sx` or `styles`. If that is the case I agree, and we should make sure we document all these components under the same section, so it would make sense to people why some components have them flattened and other don't...

Reply:
> Yep, and I am saying system in case if we are going to use it as a simply collection of system props, if we are going to treat is as a superset of CSS

From my perspective, we should go with the superset of CSS for the standalone property. We would gain an improved version of the CSS prop without the limitations of the Box. I can't think of any real downside.